src/keys/KeyManager.ts

```diff
@@ -442,6 +451,15 @@ class KeyManager {
     this.rootKeyPair = rootKeyPair;
     this.recoveryCode = recoveryCodeNew;
     this.rootCert = rootCert;
+    // Update topic about key change
+    const data: RootKeyPairChangeData = {
+      nodeId: this.getNodeId(),
+      tlsConfig: {
+        keyPrivatePem: this.getRootKeyPairPem().privateKey,
+        certChainPem: await this.getRootCertChainPem(),
+      },
+    };
+    await this.rootKeyPairChange(data);
```

Comment:
Compared to `resetRootKeyPair`, there should be an expectation that the old certificate still works...

Do you have tests included that demonstrate that GRPC clients that are still using the old certificate can still contact the the Polykey agent?

Note that this is supposed to work for agent to agent communication, but also CLI or GUI to agent communication.

The CLI is less of a problem simply because it reads the status each time, and adds the new Node ID to each call.

See the `verifyServerCertificateChain` and `verifyClientCertificateChain` methods in `src/network/utils.ts`.

This may be better done in a later PR to fix this though so we can merge this earlier, since it may be more complicated.

Reply:
This was mentioned here https://github.com/MatrixAI/js-polykey/issues/312#issuecomment-1014188218 but wasn't fleshed out, will write more details in #312.

Reply:
Wrote it up here: https://github.com/MatrixAI/js-polykey/issues/312#issuecomment-1016137891

Reply:
This will need be a new issue @emmacasolin, can't solve in this PR.

Reply:
Created here https://github.com/MatrixAI/js-polykey/issues/317